.github/workflows/spack_build.yml

```diff
@@ -141,10 +146,18 @@ jobs:
                 padded_length: False
             mirrors:
               local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
-            # spack: https://binaries.spack.io/develop
+              # If the spack buildcache fails for any readon, feel free to disable
+              spack: https://binaries.spack.io/develop
             packages:
               all:
-                require: "%gcc"
+                require: "%${{ matrix.compiler }}"
+              # https://spack.readthedocs.io/en/latest/packages_yaml.html#setting-requirements-on-virtual-specs
+              cmake:
+                # We don't need CMake GUI features for minimal container builds
+                require: "~qtgui~ncurses"
+              libevent:
+                # Building OpenSSL was causing errors
+                require: "~openssl"
```

Comment:
I was finding spack's `^` syntax in the spec to not always specify variants as I wanted. This is a more sustainable way to always enforce certain options on desired pacakges.